Source/msg.cpp

```diff
@@ -750,7 +750,7 @@ void DeltaLeaveSync(uint8_t bLevel)
 	if (!gbIsMultiplayer)
 		return;
 	if (leveltype == DTYPE_TOWN) {
-		DungeonSeeds[0] = AdvanceRndSeed();
+		DungeonSeeds[0] = GenerateSeed();

```

Comment:
It's possible for the town seeds to sync up and desync, and this property doesn't seem to matter much because the town seed is combined with uptime to produce what is highly likely to be a different set of shop options for every client.

https://github.com/diasurgical/devilutionX/blob/ad452cb6c03fd0f8ec6c2c616fb10a7f46fb5b4b/Source/stores.cpp#L2131-L2139

Maybe having a global RNG that isn't dependent on game seed would make this computation unnecessary.